src/main/java/com/example/boot/repository/RoleRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.example.boot.repository;
+
+import com.example.boot.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = ?1")
+    Optional<Role> getByName(String name);
```

Comment:
```suggestion
    Optional<Role> getByRoleName(Role.RoleName name);
```
Spring Data JPA is capable of automatically build the query in many cases based on the method name and parameters)